DNN Platform/DotNetNuke.Web/Api/Auth/ApiTokens/Repositories/ApiTokenRepository.cs

```diff
@@ -0,0 +1,184 @@
+                var sql = @"UPDATE {databaseOwner}{objectQualifier}ApiTokens
+SET IsDeleted=1
+WHERE (IsRevoked=1 OR ExpiresOn<@0)
+AND (PortalId=@1 OR @1=-1)
+AND (CreatedByUserId=@2 OR @2=-1)
+";
+                context.Execute(System.Data.CommandType.Text, sql, now, portalId, userId);
+            }
+        }
+
+        /// <inheritdoc />
+        public IPagedList<ApiToken> GetApiTokens(ApiTokenScope scope, bool includeNarrowerScopes, int portalId, int userId, ApiTokenFilter filter, string apiKey, int pageIndex, int pageSize)
+        {
+            using (var context = DataContext.Instance())
+            {
+                var wheres = new List<string>
+                    {
+                      "IsDeleted=0",
+                    };
+                if (includeNarrowerScopes)
+                {
+                    wheres.Add("Scope<=@2");
+                }
+                else
+                {
+                    wheres.Add("Scope=@2");
+                }
+
+                if (portalId > Null.NullInteger)
+                {
+                    wheres.Add("PortalId=@0");
+                }
+
+                if (userId > Null.NullInteger)
+                {
+                    wheres.Add("CreatedByUserId=@1");
+                }
+
+                switch (filter)
+                {
+                    case ApiTokenFilter.All:
+                        break;
+                    case ApiTokenFilter.Active:
+                        wheres.Add("IsRevoked=0 AND ExpiresOn > GETUTCDATE()");
+                        break;
+                    case ApiTokenFilter.Revoked:
+                        wheres.Add("IsRevoked=1");
+                        break;
+                    case ApiTokenFilter.Expired:
+                        wheres.Add("ExpiresOn <= GETUTCDATE()");
+                        break;
+                    case ApiTokenFilter.RevokedOrExpired:
+                        wheres.Add("(IsRevoked=1 OR ExpiresOn <= GETUTCDATE())");
+                        break;
+                }
+
+                var sql = string.Empty;
+                if (!string.IsNullOrEmpty(apiKey))
+                {
+                    sql = "INNER JOIN {databaseOwner}[{objectQualifier}ApiTokenKeys] atk ON atk.ApiTokenId = {objectQualifier}vw_ApiTokens.ApiTokenId ";
+                    wheres.Add("atk.TokenKey=@3");
+                }
+
+                sql += $"WHERE {string.Join(" AND ", wheres)} ORDER BY CreatedOnDate DESC";
+
+                var rep = context.GetRepository<ApiToken>();
+                return rep.Find(pageIndex, pageSize, sql, portalId, userId, (int)scope, apiKey);
+            }
+        }
+
+        /// <inheritdoc/>
+        public void SetApiTokenLastUsed(ApiTokenBase apiToken)
+        {
+            using (var context = DataContext.Instance())
+            {
+                var sql = @"UPDATE {databaseOwner}{objectQualifier}ApiTokens
+SET LastUsedOnDate=GETDATE()

```

Comment:
The documentation states that this is a UTC date.

```suggestion
SET LastUsedOnDate=GETUTCDATE()
```